pages/en/lb4/Controllers.md

```diff
@@ -156,3 +156,52 @@ class HelloController {
 - A `Repository` is LoopBack's database abstraction. See [Repositories](Repositories.html) for more.
 - `@get('/messages')` creates the `Route` for the Operation using `app.route()`.
 - `@param.query.number` adds a `number` param with a source of `query`.
+
+## Handling Errors in Controllers
+
+In order to specify errors for controller methods to throw, the class `HttpErrors` is used. `HttpErrors` can be found in the `@loopback/rest` package.
+
+The example below shows how `HttpErrors` can be used inside a controller method and how it can be tested.
+
+```js
+// the test
+import {HelloController} from 'path.to.controller'
+import {expect} from '@loopback/testlab'
+// ...
+
+// ...
+describe('Hello Controller', () => {
+  // ...
+  it('returns 412 Precondition Failed for non-positive limit', () => {
+    const controller = new HelloController();
+    try {
+      controller.list(0);
+      throw new Error('should have thrown an error');
+    } catch (err) {
+      expect(err).to.have.property('statusCode', 412);
+      expect(err.message).to.match(/precondition failed/i);
+    }
+  })
+  // ...
+})
+// ...
+```
+```js
+// the controller
+import 'HttpErrors' from '@loopback/rest'
+
+class HelloController {
+  constructor() {
+    this.messages = new Repository('messages');
```

Comment:
`this.repository` would be the correct property name.
Each controller has a property `repository`

Reply:
Controllers should use dependency injection to receive a Repository instance. Let's simplify this example and make the controller "pure" with no dependencies.

Reply:
Could you clarify what you mean by that @bajtos ? Is it right for me to assume that I should continue to not use dependency injection here?